app/modules/audioplayer/components/controls/controls.component.ts

```diff
@@ -12,12 +12,106 @@ import {PlayQueueItem} from '../../models/play_queue_item.model';
 
+  private timeTick: string;
+  private duration: string;
+
+  private timeTickPosition: string;
+  private progressBarWidth: number;
+
+  @ViewChild('audioPlayerProgressBarLine') audioPlayerProgressBarLine: ElementRef;
+  @ViewChild('audioPlayerHandle') audioPlayerProgressHandle: ElementRef;
+  @ViewChild('audioPlayerProgressBar') audioPlayerProgressBar: ElementRef;
+
   constructor() {
     this.audio = new Audio();
     this.playQueue.on('change:status', this.reactOnStatusChange, this);
+    this.timeTick = this.formatToHHMMSS(0);
+    this.duration = this.formatToHHMMSS(0);
+
+    this.timeTickPosition = '0px';
+
+    this.audio.addEventListener('canplay', () => {
+      this.duration = this.formatToHHMMSS(this.audio.duration);
+    });
+
+    this.audio.addEventListener('timeupdate', () => {
+      this.timeTick = this.formatToHHMMSS(this.audio.currentTime);
+      this.timeTickPosition = this.getTimeTickPositionFromTime(this.audio.currentTime);
+    });
+
+  }
+
+  ngAfterContentInit() {
+    let el = this.audioPlayerProgressHandle.nativeElement;
+    this.progressBarWidth = this.audioPlayerProgressBarLine.nativeElement.offsetWidth;
+
+    let start = 0;
+    let diff = 0;
+    let currentPos = 0;
+    let newPos = 0;
+
+    el.addEventListener('dragstart', (e: DragEvent) => {
+      start = e.pageX || e.clientX;
+      currentPos = this.getTimeTickPosition();
+    });
+
+    el.addEventListener('drag', (e: DragEvent) => {
+      let end: number = e.pageX || e.clientX;
+      if (end !== 0) {
+        diff = end - start;
+        newPos = (diff + currentPos);
+        this.setTimeTickPosition(newPos);
+      }
+    };
+
+    el.addEventListener('dragend', (e: DragEvent) => {
+      this.playTrackFromPosition(this.getTimeTickPosition());
+    };
+
+    this.audioPlayerProgressBarLine.nativeElement.addEventListener('click', (e: MouseEvent) => {
+      this.playTrackFromPosition(e.offsetX);
+    };
+
+    this.audioPlayerProgressBar.nativeElement.addEventListener('click', (e: MouseEvent) => {
+      this.playTrackFromPosition(e.offsetX);
+    };
+
+  };
+
+  formatToHHMMSS(input: number): string {
+    let time = new Date(null);
+    time.setSeconds(input);
+
+    // format time from hh:mm:ss to mm:ss when hh is 0
+    if (time.getHours() === 1) {
+      return time.toISOString().substr(14, 5);
+    } else {
+      return time.toISOString().substr(11, 8);
+    }
+  }
+
+  getTimeTickPositionFromTime(time: number): string {
+    return (time * this.progressBarWidth) / this.audio.duration + 'px';
+  }
+
+  getTimeTickPosition(): number {
+    let output = this.timeTickPosition.replace(/'px'/g, '');
+    return parseInt(output, 0);
+  }
+
+  setTimeTickPosition(newPos: number): void {
+    // clipping position
+    if (newPos < 0) {
+      newPos = 0;
+    } else if (newPos > this.progressBarWidth) {
+      newPos = this.progressBarWidth;
+    }
+
+    this.timeTickPosition = newPos + 'px';
```

Comment:
Why are you concatenating `px` to the timeTickPosition? Can't you do that in the template? Without the concatenating you wouldn't need the regex in the `getTimeTickPosition()` method